cpp/arcticdb/storage/azure/azure_storage.hpp

```diff
@@ -23,11 +23,7 @@
 
 namespace arcticdb::storage::azure {
 
-class AzureStorage final : public Storage<AzureStorage> {
-
-    using Parent = Storage<AzureStorage>;
-    friend Parent;
-
+class AzureStorage final : public Storage {
```

Comment:
is there a technical reason to make it `final`?

Reply:
This indicates that concrete class aren't meant to be sub-classed.

I preserved this original choice, and have defined relevant method in concrete classes similarly.